### PR TITLE
feat: implement `FileUploader` and use in resource form 🗂️

### PR DESCRIPTION
**Pull request by wflore19**
Closes [#264](https://github.com/colorstackorg/oyster/issues/264)

## Description ✏️
This PR enhances the ResourceAttachmentField component by integrating the new FileUploader component into the UI package.

- New UI for file input
- File attachment preview
- Added styling when hovering or dragging a file over
- Added an error message for oversized files & wrong file types (see potential issues below).
image

### Potential issues 🚩:

- The current `<input type='file' />` element allows users to select files of any size, but server-side validation handles the actual size limit.
- Possible Optimization: Disabling the `Save` button when the selected file exceeds the size limit OR the file type is not supported to prevent unnecessary uploads.
image

## Type of Change 🐞

-  [x] Feature - A non-breaking change which adds functionality.
-  [ ] Fix - A non-breaking change which fixes an issue.
-  [ ] Refactor - A change that neither fixes a bug nor adds a feature.
-  [ ] Documentation - A change only to in-code or markdown documentation.
-  [ ] Tests - A change that adds missing unit/integration tests.
-  [ ] Chore - A change that is likely none of the above.

## Checklist ✅

-  [x] I have done a self-review of my code.
-  [x] I have manually tested my code (if applicable).
-  [ ] I have added/updated any relevant documentation (if applicable).

### CODE REVIEW

File: apps/member-profile/app/shared/components/resource-form.tsx [hunk 1]
```diff
@@ -63,18 +63,25 @@ export function ResourceAttachmentField({
...
         id={name}
         name={name}
-        required={required}
-        type="file"
+        required
+        {...(defaultValue && {
+          initialFile: {
+            id: defaultValue.id,
+            name: defaultValue.id,
+            size: 0,
```
Comment (ramiAbdou) [inline apps/member-profile/app/shared/components/resource-form.tsx:81]: Unfortunately we're not storing the file size in the database as of right now, so we won't be able to display it when showing the initial file.

File: packages/core/src/modules/resource/resource.types.ts [hunk 2]
```diff
@@ -16,17 +16,27 @@ export type ResourceType = ExtractValue<typeof ResourceType>;
...
+      z.union([
+        z.string().trim().min(1),
+        z.unknown().transform((value) => value as File),
+      ])
+    )
+    .transform((value) => {
+      return value.filter((attachment) => {
+        return !!(attachment as File).size;
+      });
+    }),
```
Comment (ramiAbdou) [inline packages/core/src/modules/resource/resource.types.ts:39]: This is an important bit of logic, we have a clearer path to how we're going to support the "editing" workflow a little better. If the attachment is a new one, it'll be a `File`. If the attachment is an already uploaded one, then it'll be a `string`.

**Review by ramiAbdou — APPROVED** [review]
@wflore19 Thank you so much for your work here! Honestly this is a really hard component to get right, I did a lot of research when reviewing this and you hit the spot with your approach.

I thought your idea to have the dropzone be `relative` and the actual input to be `absolute` taking up the full extent of the dropzone whilst being visibly hidden was GENIUS! That was a huge key to this component being a success.

I also really loved your idea on validating the file size client-side instead of putting the stress on the server to do that. Looking back, it's definitely a no-brainer to not allow a file to be submitted that's not the right size!

Anyways, I made a few updates building on top of your initial work, check it out [here](https://github.com/colorstackorg/oyster/compare/d0b6316023fb5bb4d13785177b6b49eb2179d329..26fc50ceba1fb0be2ae04873da335812a8981b71). The main crux of these changes were making the `FileUploader` a bit more generic (ie: dynamic `accept` and `maxFileSize` props) as well as supporting the "editing" workflow. I'm happy to sync and talk through some of these changes. Also, give these changes a spin and let me know what you think or if you find any issues.

Sometime in the future, I think the next challenge with this component will be supporting multiple attachments but that's for another day haha.

Thanks again for all this work! 🚀

**Comment by wflore19** [comment]
Thanks @ramiAbdou I like the changes you've made, the new UI looks great!

**Pull request merged**